### PR TITLE
Update example for alertmanager config with templates

### LINKED ISSUE #7075: mimirtool: Templates fail to render (not defined)
#### Describe the bug
Templates fail to render when uploaded with mimirtool.

#### To Reproduce

1. Start Mimir r269-facbb92
2. Upload template file(s) to tenant with mimirtool
3. alertmanager will fail to template
  a. `ts=2024-01-08T17:41:59.644826899Z caller=dispatch.go:352 level=error component=MultiTenantAlertmanager user=devops component=dispatcher insight=true component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="slack-notifications/slack[0]: notify retry canceled due to unrecoverable error after 1 attempts: template: :1:12: executing \"\" at <{{template \"slack.title\" .}}>: template \"slack.title\" not defined"`
4. Open the template in grafana for editing
5. Click save without changing anything
6. Templating now works.

#### Expected behavior

Templating should be

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: helm via kustomize and ArgoCD
-

### PR DESCRIPTION
**Pull request by alessandro-verzicco**
#### What this PR does

Update documentation to make the `alertmanager.yml` config example coherent. In particular the example shows how to configure alertmanager with a custom alert template using mimirtool, but omits an important part: specifying the template file name in the alertmanager config. This will lead to a mis-configured alertmanager which will not send any notification and logs similar to:
```
caller=dispatch.go:353 level=error component=MultiTenantAlertmanager user=infra component=dispatcher insight=true component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="slack-infra/slack[0]: notify retry canceled due to unrecoverable error after 1 attempts: template: :1:12: executing \"\" at <{{template \"standard_alert\" .}}>: template \"standard_alert\" not defined"
```

#### Which issue(s) this PR fixes or relates to

Fixes #7075

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by tacole02 — APPROVED** [review]

**Review by dimitarvdimitrov — APPROVED** [review]

File: docs/sources/mimir/manage/tools/mimirtool.md [hunk 1]
```diff
@@ -144,6 +144,8 @@ route:
   group_by: ["example_groupby"]
 receivers:
   - name: "example_receiver"
+templates:
+  - example_alertmanager_template.tpl
```
Comment (dimitarvdimitrov) [inline docs/sources/mimir/manage/tools/mimirtool.md:148]:
```suggestion
  - example_alertmanager_template.tpl
  
```

File: docs/sources/mimir/manage/tools/mimirtool.md [hunk 2]
```diff
@@ -144,6 +144,9 @@ route:
...
+
```
Comment (dimitarvdimitrov) [inline docs/sources/mimir/manage/tools/mimirtool.md:149]:
```suggestion
```

File: docs/sources/mimir/manage/tools/mimirtool.md (hunk 1 above)
Comment (dimitarvdimitrov) [inline docs/sources/mimir/manage/tools/mimirtool.md:148]:
```suggestion
  - example_template.tpl
```

Comment (dimitarvdimitrov) [reply docs/sources/mimir/manage/tools/mimirtool.md:148]: trying to trigger CI again

**Comment by dimitarvdimitrov** [comment]
sorry, can you rebase the branch onot grafana:main and push again? I don't know how else to try to get the CI to run. I can't merge the PR unless CI passes

**Comment by alessandro-verzicco** [comment]
done

**Comment by dimitarvdimitrov** [comment]
now i know what it was. The new merge "experience ✨" in GitHub doesn't have the "Approve and run" button. I had to switch to the old one to approve the CI. Now CI is running. I'll merge on green. Thanks again!

**Pull request merged**